cityofaustin/dts-maximo-reporting
# Maximo Work Orders to Socrata

Part of https://github.com/cityofaustin/atd-data-tech/issues/16656

A script that runs a query against the Maximo data warehouse oracle DB and then upserts the data over to the open data portal. 

## Testing
We can test this now or wait for the airflow PR?

1. fill in the `env_template` (maximo details in 1pass: `Maximo Data warehouse`)
2. git clone and cd to this directory
3. `$ docker build . -t dts-maximo-reporting:production`
4. `$ docker run -it --env-file env_file dts-maximo-reporting:production /bin/bash`
5. `$ python etl/work_orders_to_socrata.py`
6. Check if for errors and you should get something like:

> 2024-05-15 15:20:51,927 INFO: Getting work orders with start: 05/08/2024, end: 05/15/2024
> 2024-05-15 15:20:54,906 INFO: 1139 work orders found
> 2024-05-15 15:20:56,853 INFO: {'Errors': 0, 'Rows Created': 0, 'Rows Updated': 1139, 'Rows Deleted': 0}

That last line is the response from Socrata 👆 

[Dataset is here
](https://datahub.austintexas.gov/Transportation-and-Mobility/Street-and-Bridge-Operations-Work-Orders/hjym-dxqr)
